R/priorsense.R

```diff
@@ -51,13 +51,19 @@ create_priorsense_data.brmsfit <- function(x, ...) {
 
 #' @exportS3Method priorsense::log_lik_draws
 log_lik_draws.brmsfit <- function(x, ...) {
+  nchains <- nchains(x)
+  niters <- niterations(x)
   log_lik <- log_lik(x, ...)
-  log_lik <- posterior::as_draws_array(log_lik)
+  nobs <- length(log_lik) / (nchains * niters)
```

Comment:
This is dangerous if `...` already subsets draws for some reason. So you need to keep track of `ndraws` and `draw_ids` arguments as passed to `log_lik` and only store chains information if these arguments were not specified.

Reply:
Very good point. I'll revise this, and update the PR

Reply:
I now changed it to check if `nrow(log_lik(x, ...)) < niterations(x) * nchains(x)` and if so, then just set `nchains <- 1` and `niters <- nrow(log_lik(x, ...))`. I think this is simpler than checking for given arguments in `...`, which I tried in the previous commit.